pulsar-broker/src/main/java/org/apache/pulsar/broker/service/Producer.java

```diff
@@ -246,18 +246,26 @@ public boolean checkAndStartPublish(long producerId, long sequenceId, ByteBuf he
 
     private void publishMessageToTopic(ByteBuf headersAndPayload, long sequenceId, long batchSize, boolean isChunked,
                                        boolean isMarker) {
-        topic.publishMessage(headersAndPayload,
-                MessagePublishContext.get(this, sequenceId, msgIn,
-                        headersAndPayload.readableBytes(), batchSize,
-                        isChunked, System.nanoTime(), isMarker));
+        MessagePublishContext messagePublishContext =
+                MessagePublishContext.get(this, sequenceId, msgIn, headersAndPayload.readableBytes(),
+                        batchSize, isChunked, System.nanoTime(), isMarker);
+        if (this.cnx.getBrokerService().getInterceptor() != null){
+            this.cnx.getBrokerService().getInterceptor()
+                    .onMessagePublish(this, headersAndPayload, messagePublishContext);
+        }
+        topic.publishMessage(headersAndPayload, messagePublishContext);
     }
 
     private void publishMessageToTopic(ByteBuf headersAndPayload, long lowestSequenceId, long highestSequenceId,
                                        long batchSize, boolean isChunked, boolean isMarker) {
-        topic.publishMessage(headersAndPayload,
-                MessagePublishContext.get(this, lowestSequenceId,
-                        highestSequenceId, msgIn, headersAndPayload.readableBytes(), batchSize,
-                        isChunked, System.nanoTime(), isMarker));
+        MessagePublishContext messagePublishContext = MessagePublishContext.get(this, lowestSequenceId,
+                highestSequenceId, msgIn, headersAndPayload.readableBytes(), batchSize,
+                isChunked, System.nanoTime(), isMarker);
+        if (this.cnx.getBrokerService().getInterceptor() != null){
```

Comment:
If we can have a `InterceptorDisabled` implementation, we can avoid many `if` checks like here.

Reply:
`BrokerInterceptorDisabled` is already implemented. Initialize the interceptor as a `BrokerInterceptorDisabled` if no interceptors are configured.